source/numerics/solution_transfer.inst.in

```diff
@@ -16,9 +16,56 @@
 for (VEC : VECTOR_TYPES; deal_II_dimension : DIMENSIONS;
      deal_II_space_dimension : SPACE_DIMENSIONS)
   {
+#if deal_II_dimension <= deal_II_space_dimension
+    template class Legacy::
+      SolutionTransfer<deal_II_dimension, VEC, deal_II_space_dimension>;
+#endif
+  }
+
+for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS)
+  {
 #if deal_II_dimension <= deal_II_space_dimension
     template class SolutionTransfer<deal_II_dimension,
-                                    VEC,
+                                    ::dealii::Vector<double>,
                                     deal_II_space_dimension>;
+    template class SolutionTransfer<deal_II_dimension,
+                                    ::dealii::BlockVector<double>,
+                                    deal_II_space_dimension>;
```

Comment:
You're listing specific vector types here. Why not leave it a loop over all vector types? Does the class not work for specific vector types? If so, which are these?